packages/transaction-controller/src/utils/first-time-interaction-api.ts

```diff
@@ -0,0 +1,75 @@
+import { createModuleLogger, type Hex } from '@metamask/utils';
+
+import { FirstTimeInteractionError } from '../errors';
+import { projectLogger } from '../logger';
+
+export const BASE_URL = 'https://accounts.api.cx.metamask.io';
+const FAILED_TO_PARSE_MESSAGE = 'Failed to parse account address relationship.';
+
+const log = createModuleLogger(projectLogger, 'first-time-interaction-api');
+
+export type FirstTimeInteractionRequest = {
+  /** Chain ID of the transaction. */
+  chainId: Hex;
+
+  /** Recipient of the transaction. */
+  to?: string;
+
+  /** Sender of the transaction. */
+  from: string;
+};
+
+export type FirstTimeInteractionResponse = {
+  isFirstTimeInteraction?: boolean;
+};
+
+/**
+ * Get the first time interaction count for an account.
+ * @param request - The request to get the first time interaction count for.
+ * @returns The first time interaction count for the account.
+ */
+export async function getFirstTimeInteraction(
+  request: FirstTimeInteractionRequest,
+): Promise<FirstTimeInteractionResponse> {
+  const url = await getFirstTimeInteractionUrl(request);
+
+  log('Sending request', url, request);
+
+  const response = await fetch(url, {
+    method: 'GET',
+  });
+
+  const responseJson = await response.json();
+
+  log('Received response', responseJson);
+
+  if (responseJson.error) {
+    const { message, code } = responseJson.error;
+
+    if (message === FAILED_TO_PARSE_MESSAGE) {
```

Comment:
Is there an error code we could use instead to keep it less coupled?

Reply:
After discussion with Micheal, I've added `isFirstTimeInteractionDisabled` for some cases, so UI will ignore when `isFirstTimeInteractionDisabled` is settled hence I've removed this part but for other case (where the status is `204`) I've coupled it with code.